File: lua/collatz-conjecture/collatz-conjecture.lua
return function(t)
  local function collatz(n)
    if (n <= 0) then
      error("Only positive numbers are allowed")
    elseif (n == 1) then
      return 0
    end
    return 1 + collatz(n % 2 == 0 and n / 2 or 3 * n + 1)
  end
  return collatz(t)
end

File: javascript/collatz-conjecture/collatz-conjecture.js
export const steps = (n) => {
  if (n <= 0) {
    throw new Error('Only positive integers are allowed');
  }
  if (n == 1) {
    return 0;
  }
  return 1 + steps(n % 2 === 0 ? n / 2 : n * 3 + 1);
};
